src/ui/public/crypto/sha256.js

```diff
@@ -0,0 +1,194 @@
+// ported from https://github.com/spalger/sha.js/blob/6557630d508873e262e94bff70c50bdd797c1df7/sha256.js
```

Comment:
Why strive for a non-cryptographic hash function? Why is the speed of sha256 a concern? I would rather use something that is commonly used and understood, and I think that shaX is it.

[git uses sha1](http://stackoverflow.com/a/28792805/296172) and most download verifications use sha or md5.